### PR TITLE
fix #1110

### LINKED ISSUE #1110: Freezed Lint reports unneeded private empty constructor
**Describe the bug**
Freezed Lint reports to create an private empty constructor when a static class method or variable is added.

**To Reproduce**
```dart
@freezed
class Example with _$Example {
  static const int a = 0; // <- causes lint rule to trigger

  static String b = 0.toString(); // <- causes lint rule to trigger

  static int c() => 0; // <- causes lint rule to trigger

  const factory Example({
    required String e,
  }) = _Example;
}
```

**Expected behavior**
Lint rule should not apply.

### PR DESCRIPTION
**Pull request by SunlightBro**
- add custom_lint.log to .gitignore
- fix MissingPrivateEmptyCtor
- add test-case inside example
- prepare version 0.0.6 & CHANGELOG

### CODE REVIEW

File: packages/freezed_lint/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.0.6 - 2024-07-16
```
Comment (rrousselGit) [inline packages/freezed_lint/CHANGELOG.md:1]:
```suggestion
## Unreleased fix
```

File: packages/freezed_lint/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+
+- fix false positive `freezed_missing_private_empty_constructor` if accessor/method/field is static
```
Comment (rrousselGit) [inline packages/freezed_lint/CHANGELOG.md:3]:
```suggestion
- fix false positive `freezed_missing_private_empty_constructor` if accessor/method/field is static (thanks to @SunlightBro)
```

File: packages/freezed_lint/pubspec.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: freezed_lint
-version: 0.0.5
+version: 0.0.6
```
Comment (rrousselGit) [inline packages/freezed_lint/pubspec.yaml:2]:
```suggestion
version: 0.0.5
```

**Comment by SunlightBro** [comment]
using **stable** or **beta** `flutter analyze .` has no issues.

but on master the main issue (_I think_) is in custom_lint_core/lib/src/lint_codes.dart,
because analyzer/src/dart/error/lint_codes.dart now also defines `LintCode`.

**Comment by rrousselGit** [comment]
Do you mind adding the necessary import alias or `show/hide`?

**Comment by SunlightBro** [comment]
Requires one change in [custom_lint_core/lib/src/lint_rule.dart
](https://github.com/invertase/dart_custom_lint/blob/c67d3527a215fdb06b95c2cc1b2c2633c9e55af7/packages/custom_lint_core/lib/src/lint_rule.dart#L7)

```diff
  import 'package:analyzer/dart/ast/ast.dart';
- import 'package:analyzer/error/error.dart';
+ import 'package:analyzer/error/error.dart' hide LintCode;
  import 'package:analyzer/error/listener.dart';
```

File: packages/freezed_lint/lib/src/missing_private_empty_ctor.dart [hunk 4]
```diff
@@ -29,9 +29,10 @@ class MissingPrivateEmptyCtor extends DartLintRule {
       final annotation = freezedAnnotationChecker.hasAnnotationOfExact(element);
       if (!annotation) return;
 
-      final methods = element.methods;
-      final fields = element.fields;
-      final accessors = element.accessors;
+      final methods = element.methods.where((method) => !method.isStatic);
+      final fields = element.fields.where((field) => !field.isStatic);
+      final accessors =
```
Comment (snapsl) [inline packages/freezed_lint/lib/src/missing_private_empty_ctor.dart:34]: Nice catch to add accessors! Thank you for the quick fix @SunlightBro

**Comment by rrousselGit** [comment]
TY!

**Comment by lishaduck** [comment]
@rrousselGit, can you publish this? Running `custom_lint` in one of my projects errors with `[path-to-freezed] 'LintCode' is imported from both [path-to-custom_lint_core] [path-to-analyzer]. Failed to start plugins The request analysis.setContextRoots failed with the following error: RequestErrorCode.PLUGIN_ERROR Bad state: Failed to start the plugins.` without this PR, but pulling it in as a git dependency works fine. I'm as up-to-date as one can be on the stable channel.
(I'm not sure why this fixes it, I have another project on the stable channel that also depends on `analyzer@6.7.0`, `custom_lint_core@0.6.5` and `freezed_lint@0.0.5` which works fine, but 🤷‍♂️)

**Pull request merged**